integration-cli/docker_cli_daemon_test.go

```diff
@@ -828,3 +828,66 @@ func TestDaemonUnixSockCleanedUp(t *testing.T) {
 
 	logDone("daemon - unix socket is cleaned up")
 }
+
+func TestDaemonwithwrongkey(t *testing.T) {
+	type Config struct {
+		Crv string `json:"crv"`
+		D   string `json:"d"`
+		Kid string `json:"kid"`
+		Kty string `json:"kty"`
+		X   string `json:"x"`
+		Y   string `json:"y"`
+	}
+
+	os.Remove("/etc/docker/key.json")
+	k, err := libtrust.GenerateECP256PrivateKey()
+	if err != nil {
+		t.Fatalf("Error generating private key: %s", err)
+	}
+	if err := os.MkdirAll(filepath.Join(os.Getenv("HOME"), ".docker"), 0755); err != nil {
+		t.Fatalf("Error creating .docker directory: %s", err)
+	}
+	if err := libtrust.SaveKey(filepath.Join(os.Getenv("HOME"), ".docker", "key.json"), k); err != nil {
+		t.Fatalf("Error saving private key: %s", err)
+	}
+	d := NewDaemon(t)
+	if err := d.Start(); err != nil {
+		t.Fatalf("Failed to start daemon: %v", err)
+	}
+
+	if err := d.Stop(); err != nil {
+		t.Fatalf("Could not stop daemon: %v", err)
+	}
+
+	config := &Config{}
+	bytes, err := ioutil.ReadFile("/etc/docker/key.json")
```

Comment:
I imagine this is following the pattern set in the key migration test.  This should not be necessary since when "/etc/docker/key.json" is removed the daemon should create a new one.